app/js/enhancedDate.js

```diff
@@ -1,5 +1,156 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function (date) {
+// still maintains access to all globals
+  
+  var dateMethods = {};
+  var myDate = date;
+  var dateString = Date(year, month, day, hours, minutes, seconds, milliseconds);
+  var newDate = new Date();
+  var epochDate = newDate.getTime();
+  
+//Stores the passed-in date.  
+  dateMethods.checkDate = function () {
+      
+//Do some type-checking to make sure you have a valid type. 
+//Takes milliseconds after epoch or a date object. 
+//Do some type-checking to make sure you have a valid type.
+    
+      if (date){
+        if(myDate instanceof Date)
+        {
+        return true;
+        
+        } else if (typeof myDate === 'number' && (myDate > 0 && myDate <= epochDate))
+        {
+        return true;
+        
+        } else {
+        return false;  
+        } 
+      }
+
+//If no value is passed, default to now.   
+      else {
+        myDate = newDate;
+        return true;
+      }
+  };
+
+
+//Returns either milliseconds after epoch by default or a date object if true is passed as an argument.            
+  dateMethods.getDate = function (){
+    dateMethods.checkDate();
+    
+    if (dateMethods.checkDate === true) {
+        return myDate;
+    }
+  };
+
+//Returns the full day name as a string, e.g. "Monday", "Tuesday", etc.  
+  dateMethods.getDayName = function () {
+    dateMethods.checkDate();
+    
+   if (dateMethods.checkDate === true) {
+      switch (myDate.getDay()) {
+        case 0:
+            day = "Sunday";
+            break;
+        case 1:
+            day = "Monday";
+            break;
+        case 2:
+            day = "Tuesday";
+            break;
+        case 3:
+            day = "Wednesday";
+            break;
+        case 4:
+            day = "Thursday";
+            break;
+        case 5:
+            day = "Friday";
+            break;
+        case 6:
+            day = "Saturday";
+            break;
+      }
+    } 
+    
+    return;
```

Comment:
You want to return day here: `return day;` If you just use the return keyword by itself, it will return undefined. You've set day, so now you just need to return it.